### PR TITLE
Use the official Guacamole Docker image

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request modifies the `docker-compose.yml` file to use [the official `guacamole/guacamole` Docker image](https://hub.docker.com/r/guacamole/guacamole) instead of [our custom `cisagov/guacamole` image](https://hub.docker.com/r/cisagov/guacamole).  We were using the custom Guacamole image because the official one lacked support for HTTP header authentication, but as of version 1.3.0 the official image includes the necessary support - added via [apache/guacamole#548](https://github.com/apache/guacamole-client/pull/548).

While I was mucking around in here, I went ahead and cleaned up a few things:
* Reduced the `version` in the `docker-compose.yml` file to the minimum required
* Deleted a duplicate `secrets` directory
* Added a `requirements-dev.txt` file to install `semver` for versioning

## 💭 Motivation and Context ##

We should use the official Docker image now that it is possible, if only to avoid the hassle of maintaining our own custom image.

## 🧪 Testing ##

I transferred this `docker-compose.yml` file to env0 in our staging COOL environment and verified that it worked as expected.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Excellent- thanks for taking care of this! 🥑 

We should test if this new image fixes https://github.com/cisagov/cool-system/issues/123.

**Comment by jsf9k** [comment]
> We should test if this new image fixes [cisagov/cool-system#123](https://github.com/cisagov/cool-system/issues/123).

Ah, yes.  I had forgotten about that.

FYI, I will build a new AMI once this PR is merged.

**Review by hillaryj — APPROVED** [review]
Great work! 🥇

File: docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-version: "3.7"
+version: "3.2"
```
Comment (mcdonnnj) [inline docker-compose.yml]: Not to nitpick, but does this actually use any `3.2` features?

**Review by mcdonnnj — APPROVED** [review]
Thanks for tracking this and getting us updated! 💪 I had a minor question that doesn't block approval from me.

File: docker-compose.yml (hunk 1 above)
Comment (jsf9k) [reply docker-compose.yml]: I think you're right @mcdonnnj.  I was thinking that secrets was added in 3.2, but it was actually added in 3.1.  Do you concur that the version should be 3.1?

Comment (jsf9k) [reply docker-compose.yml]: See commit 75388d5.

Comment (mcdonnnj) [reply docker-compose.yml]: `3.1` is what I was thinking as well.

**Review by mcdonnnj — APPROVED** [review]
image

**Pull request merged**